docs/book/src/usage/ci.md

```diff
@@ -0,0 +1,136 @@
+
+#### Basic usage
+
+Checkout the repository code, install the toolchain, Marker, and start linting.
+
+```yml
+jobs:
+  rust-marker-lints:
+    runs-on: ubuntu-latest
+    env:
+      RUSTFLAGS: --deny warnings
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions-rust-lang/setup-rust-toolchain@v1
+      - uses: rust-marker/marker@v0.2
+```
+
+#### Advanced usage
+
+If you need something more than just the `cargo marker` command, you may use the action to only install Marker and then manually run the `cargo marker` command, just like in your local dev environment.
+
+Here is an example of how you could limit the set of crates that you want to lint. Refer to `cargo marker --help` for a full list of available options.
+
+```yml
+jobs:
+  rust-marker-lints:
+    runs-on: ubuntu-latest
+    env:
+      RUSTFLAGS: --deny warnings
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions-rust-lang/setup-rust-toolchain@v1
+      - uses: rust-marker/marker@v0.2
+        with:
+          install-only: true
+      - run: cargo marker -- -p crate-foo -p crate-bar
+```
+
+If you have an example of advanced usage of `cargo marker` command that you have to repeat in your CI template again and again consider opening a [new issue] in our repository. We will be glad to hear any suggestions about extending the inputs for the GitHub Action for your use case.
+
+
+### GitHub-managed runners
+
+The action is cross-platform. It supports Windows, Linux and MacOS. It is tested on all [OS images supported by managed GitHub Actions runners].
+
+If GitHub releases a newer OS image version it is very likely that this action will still "just work" on it. We make sure to add new OS images that GitHub-managed runners support to our CI test suite, as well as remove support for the old ones following the GitHub Actions's OS images deprecation cadance.
+
+### Self-hosted runners
+
+The action must work on self-hosted runners out of the box as well. We don't test all possible operating systems on our CI, but if your self-hosted runners use the OS version from the range of [OS images supported by managed GitHub Actions runners] or other compatible distributions, then it should work. If you have some exotic OS setup which breaks our GitHub Action we will be interested to hear about that in a [new issue].
+
+## Other CI systems
+
+If you don't use GitHub Actions CI, you can still benefit from the pre-made installation scripts that automate the downloading of the pre-compiled binaries on CI for you.
+
+These ones slightly differ from the scripts mentioned in the [installation chapter](installation.md#download-pre-compiled-binaries-recommended). They are more verbose for additional readability in the CI templates, and they also contain additional options to retry spurious network errors for stability on CI.
```

Comment:
Super NIT: I think the "These ones" could be confusing, since we talked about installation scripts earlier.

```suggestion
These curl commands differ slightly from the scripts mentioned in the [installation chapter](installation.md#download-pre-compiled-binaries-recommended). They are more verbose for additional readability in the CI templates, and they also contain additional options to retry spurious network errors for stability on CI.
```